src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/TypeThreadStaticIndexNode.cs

```diff
@@ -15,19 +15,17 @@ public class TypeThreadStaticIndexNode : DehydratableObjectNode, ISymbolDefiniti
         private MetadataType _type;
         private ThreadStaticsNode _inlinedThreadStatics;
 
-        public TypeThreadStaticIndexNode(MetadataType type)
+        public TypeThreadStaticIndexNode(MetadataType type, ThreadStaticsNode inlinedThreadStatics)
         {
             _type = type;
-        }
-
-        public TypeThreadStaticIndexNode(ThreadStaticsNode inlinedThreadStatics)
-        {
             _inlinedThreadStatics = inlinedThreadStatics;
         }
 
+        public bool IsInlined => _inlinedThreadStatics != null;
+
         public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
         {
-            sb.Append(_type != null ? nameMangler.NodeMangler.ThreadStaticsIndex(_type) : "_inlinedThreadStaticsIndex");
+            sb.Append(nameMangler.NodeMangler.ThreadStaticsIndex(_type));
```

Comment:
The key change. We used to emit just one index node for the entire storage block and were reusing it for every type involved. 
Now we are back to emitting individual index nodes for all types that are known to have threadstatics. This allows natvis to know the offsets of each type's storage within the block.